instance/management/commands/archive_instances.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app - Archive one or more instances by their domains
+"""
+
+# Imports #####################################################################
+
+import logging
+
+from django.core.management.base import BaseCommand
+
+from instance.models.openedx_instance import OpenEdXInstance
+
+LOG = logging.getLogger(__name__)
+
+# Classes #####################################################################
+
+
+class Command(BaseCommand):
+    """
+    archive_instances management command class
+    """
+    help = 'Archive instances specified by their internal LMS domain.'
+
+    def add_arguments(self, parser):
+        """
+        Add mutually exclusive required arguments --domains and --file to parser.
+        """
+        domains_source_group = parser.add_mutually_exclusive_group(required=True)
+        domains_source_group.add_argument(
+            '--domains', help='Comma-separated list of domains'
+        )
+        domains_source_group.add_argument(
+            '--file', help='File containing newline-separated list of instance domains'
+        )
+
+    def handle(self, *args, **options):
+        """
+        Archive instances from a list of domains or from a file.
+        """
+        domains = options['domains']
+        infile = options['file']
+
+        if infile:
+            with open(infile, 'r') as f:
+                domains = [line.strip() for line in f.readlines()]
```

Comment:
@john2x, this will read all the lines of the file into a list and then iterate over it. It is better to use
```suggestion
                domains = [line.strip() for line in f]
```
as it will directly read the line from the file and construct the list.

Reply:
@lgp171188 Thanks, will keep that in mind.